packages/bot-cleaning/src/MarketCleaner.ts

```diff
@@ -151,10 +154,14 @@ export class MarketCleaner {
         const outboundTkn =
           ba == "bids" ? semibook.market.quote : semibook.market.base;
 
-        const takerWants = new Big(1).div(
-          new Big(10).pow(outboundTkn.decimals)
+        let takerWants = new Big(1).div(new Big(10).pow(outboundTkn.decimals));
+        let takerGives = inboundTkn.fromUnits(
+          takerWants.mul(offer.price!).toString()
         );
-        const takerGives = new Big(1).div(new Big(10).pow(inboundTkn.decimals));
+        if (outboundTkn.decimals > inboundTkn.decimals) {
+          takerGives = new Big(1).div(new Big(10).pow(inboundTkn.decimals));
+          takerWants = takerGives.mul(new Big(1).div(offer.price!).round());
+        }
```

Comment:
Not sure i understand why you switch decimals

Reply:
I'm not switching decimals ! 

The logic I tried to implement is to always force the maker contract to retrieve at least one unit of the outbountTkn. (One unit of a token = 10^-decimals)

In the case that `outboundTkn` have bigger decimals that `inboundTkn` the logic is to set the minimal volume of `takerGives` and compute the minimal volume of `takerWants` accordingly.

For example:
- outboundTkn is WETH and have 18 decimals
- inboundTkn is USDC and have 6 decimals
- if I set `takerWants` as `10^-18` and I then multiply this by the price to compute `takerGives` I will end up with a value smaller than `10^-6 `which doesn't make sense.
- Instead I set `takerGives` to `10^-6` and I compute the minimal `takerWants` accordingly.